analysisstore/client/commands.py

```diff
@@ -162,14 +159,42 @@ def update_analysis_header(self, query, update):
     def update_analysis_tail(self, query, update):
         pass
 
-    def find_analysis_header(self, **kwargs):
-        pass
+    def find_analysis_header(self, as_json=False, **kwargs):
+        r = requests.get(self.aheader_url, params=ujson.dumps(kwargs))
+        r.raise_for_status()
+        content = ujson.loads(r.text)
+        if as_json:
+            return r.text
+        else:        
+            for c in content:
+                yield Document('AnalysisHeader', c)
 
-    def find_analyis_tail(self, **kwargs):
-        pass
+    def find_analyis_tail(self, as_json=False, **kwargs):
+        r = requests.get(self.atail_url, params=ujson.dumps(kwargs))
+        r.raise_for_status()
+        content = ujson.loads(r.text)
+        if as_json:
+            return r.text
+        else:        
+            for c in content:
+                yield Document('AnalysisTail', c)
 
-    def find_data_reference_header(self, **kwargs):
-        pass
+    def find_data_reference_header(self, as_json=False, **kwargs):
+        r = requests.get(self.dref_header_url, params=ujson.dumps(kwargs))
+        r.raise_for_status()
+        content = ujson.loads(r.text)
+        if as_json:
+            return r.text
+        else:        
+            for c in content:
+                yield Document('DataReferenceHeader', c)
 
-    def find_data_reference(self, **kwargs):
-        pass
+    def find_data_reference(self, as_json=False, **kwargs):
+        r = requests.get(self.dref_url, params=ujson.dumps(kwargs))
+        r.raise_for_status()
+        content = ujson.loads(r.text)
+        if as_json:
+            return r.text
+        else:        
+            for c in content:
+                yield Document('DataReference', c)
```

Comment:
Yup, @klauer already suggested this. I will clean it up once I get it all working. There are quite a few things that are repeated and must belong in command_utils.py

PS: I will also separate the client from the server once everything is functional and tested